src/basic/mkdir.c

```diff
@@ -136,34 +136,54 @@ int mkdir_parents_internal(const char *prefix, const char *path, mode_t mode, mk
                 if (prefix && path_startswith(prefix, t))
                         continue;
 
-                r = _mkdir(t, mode);
-                if (r < 0 && r != -EEXIST)
-                        return r;
+                if (uid == UID_INVALID && gid == UID_INVALID && flags == 0) {
```

Comment:
should use uid_is_valid() probably

Reply:
Thanks, fixed in https://github.com/systemd/systemd/pull/16485